main.tf

```diff
@@ -290,3 +293,44 @@ resource "google_storage_bucket_iam_member" "api_s_account_storage_admin_on_proj
 
   depends_on = ["google_project_service.project_services"]
 }
+
+/******************************************
+  compute.networkUser role granted to API and GKE service accounts for GKE on shared VPC 
+ *****************************************/
+resource "google_project_iam_binding" "gke_shared_vpc_project" {
+  count = "${local.gke_shared_vpc_enabled && length(compact(var.shared_vpc_subnets) == 0) ? 1 : 0}"
+
+  role    = "roles/compute.networkUser"
+  project = "${var.shared_vpc}"
+  members = ["${compact(list(local.api_s_account_fmt, local.gke_s_account_fmt))}"]
+
+  depends_on = ["google_project_service.project_services"]
+}
+
+/******************************************
+  compute.networkUser role granted to API and GKE service accounts for GKE on shared VPC subnets
+ *****************************************/
+resource "google_compute_subnetwork_iam_binding" "gke_shared_vpc_subnets" {
+  count = "${local.gke_s_account_fmt && length(compact(var.shared_vpc_subnets) != 0) ? length(var.shared_vpc_subnets) : 0}"
```

Comment:
`local.gke_s_account_fmt` looks wrong?

Reply:
yup, planning to add tests by hopefully the end of the week. 

and yup, that's super wrong.